model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -798,26 +798,59 @@ def _determine_local_domains(self):
         )
         self._end_vertex_halo = self._grid.end_index(vertex_domain(h_grid.Zone.HALO))
 
-    def set_timelevels(self, nnow, nnew):
-        #  Set time levels of ddt_adv fields for call to velocity_tendencies
-        if self._config.itime_scheme == TimeSteppingScheme.MOST_EFFICIENT:
-            self.ntl1 = nnow
-            self.ntl2 = nnew
-        else:
-            self.ntl1 = 0
-            self.ntl2 = 0
+    def update_time_levels_for_velocity_tendencies(
+        self,
+        diagnostic_state_nh: dycore_states.DiagnosticStateNonHydro,
+        at_first_substep: bool,
+        at_initial_substep: bool,
+    ) -> Literal[0, 1]:
+        """
+        Set time levels of ddt_adv fields for call to velocity_tendencies.
+
+        When using `TimeSteppingScheme.MOST_EFFICIENT` (itime_scheme=4 in ICON Fortran),
+        `ddt_w_adv_pc.current` (advection term in vertical momentum equation in
+        predictor step) is not computed in the predictor step of each substep.
+        Instead, the advection term computed in the corrector step during the
+        previous substep is reused for efficiency (except, of course, in the
+        very first substep of the initial time step).
+        `ddt_v_apc.current` (advection term in horizontal momentum equation in
+        predictor step) is only computed in the predictor step of the first
+        substep and the advection term in the corrector step during the previous
+        substep is reused for `ddt_v_apc.current` from the second substep onwards.
+        Additionally, in this scheme the predictor and corrector outputs are kept
+        in separated elements of the pair (.current for the predictor and .next
```

Comment:
```suggestion
        in separate elements of the pair (.current for the predictor and .next
```

Reply:
Just an idea: as the naming of the pair elements is flexible we could even have a 
```python
class PredictorCorrectorPair:(Pair[T]):
    predictor: T = Pair.first
    corrector: T = Pair.frozen_second
```
 and then use
`ddt_w_adv_pc.predictor` and `ddt_w_adv_pc.corrector`

Reply:
It is a nice name! Correctly conveying why we need two buffers. I will use this name.